File: qqbot-persistence/src/main/java/com/mieai/qqbot/persistence/plugin/PluginArtifactRowMapper.java
package com.mieai.qqbot.persistence.plugin;

import com.mieai.qqbot.persistence.internal.UtcTimestampCodec;
import java.sql.ResultSet;
import java.sql.SQLException;
import org.springframework.jdbc.core.RowMapper;

final class PluginArtifactRowMapper implements RowMapper<PluginArtifact> {
    @Override
    public PluginArtifact mapRow(ResultSet rs, int rowNum) throws SQLException {
        return new PluginArtifact(rs.getString("plugin_id"), rs.getString("name"), rs.getString("version"),
                rs.getString("api_compatibility"), rs.getString("file_name"), rs.getString("sha256"),
                rs.getString("entrypoint"), rs.getString("status"), rs.getInt("enabled") != 0,
                UtcTimestampCodec.parse(rs.getString("created_at")),
                UtcTimestampCodec.parse(rs.getString("updated_at")));
    }
}

File: qqbot-persistence/src/main/kotlin/com/mieai/qqbot/persistence/plugin/PluginArtifactRowMapper.kt
package com.mieai.qqbot.persistence.plugin

import com.mieai.qqbot.persistence.internal.UtcTimestampCodec
import java.sql.ResultSet
import org.springframework.jdbc.core.RowMapper

internal class PluginArtifactRowMapper : RowMapper<PluginArtifact> {
    override fun mapRow(resultSet: ResultSet, rowNumber: Int): PluginArtifact = PluginArtifact(
        resultSet.getString("plugin_id"),
        resultSet.getString("name"),
        resultSet.getString("version"),
        resultSet.getString("api_compatibility"),
        resultSet.getString("file_name"),
        resultSet.getString("sha256"),
        resultSet.getString("entrypoint"),
        resultSet.getString("status"),
        resultSet.getInt("enabled") != 0,
        UtcTimestampCodec.parse(resultSet.getString("created_at")),
        UtcTimestampCodec.parse(resultSet.getString("updated_at")),
    )
}
